shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -445,6 +473,58 @@ - (UITextRange*)lineEnclosingPosition:(UITextPosition*)position {
 
 @end
 
+#pragma mark - FlutterTextSelectionRect
+
+@implementation FlutterTextSelectionRect
+
+@synthesize rect = _rect;
+@synthesize writingDirection = _writingDirection;
+@synthesize containsStart = _containsStart;
+@synthesize containsEnd = _containsEnd;
+@synthesize isVertical = _isVertical;
+
++ (instancetype)selectionRectWithRectAndInfo:(CGRect)rect
+                            writingDirection:(NSWritingDirection)writingDirection
+                               containsStart:(BOOL)containsStart
+                                 containsEnd:(BOOL)containsEnd
+                                  isVertical:(BOOL)isVertical {
+  return [[FlutterTextSelectionRect alloc] initWithRectAndInfo:rect
```

Comment:
Incorrect memory management, you need to return autoreleased objects from functions like this.

Reply:
Gotcha. Basically anywhere I'm calling `alloc`, I also need to `autorelease`, right? I tried to find all the places that might be in my changes.